modules/catalog/src/main/java/org/apache/ignite/internal/catalog/descriptors/CatalogHashIndexDescriptorSerializers.java

```diff
@@ -43,21 +46,22 @@ static class HashIndexDescriptorSerializerV1 implements CatalogObjectSerializer<
         public CatalogHashIndexDescriptor readFrom(CatalogObjectDataInput input) throws IOException {
             int id = input.readVarIntAsInt();
             String name = input.readUTF();
-            long updateToken = input.readVarInt();
+            long updateTimestampLong = input.readVarInt();
```

Comment:
I do not understand, why do you change `HashIndexDescriptorSerializerV1`? You brake compatibility with this changes. As far as I can understand, you only need to change `HashIndexDescriptorSerializerV2` 
But please check with sql folks regarding the correct solution.

Reply:
This doesn't imply any changes in underlying storage, and caused only by the changes in the object. Now the `long` taken from storage is interpreted in another way: it's converted into timestamp, but this doesn't break anything.

Reply:
We agreed to fix this by reading old token as initial timestamp.